lib/json-schema/validator.rb

```diff
@@ -134,6 +134,13 @@ def load_ref_schema(parent_schema, ref)
       schema_uri = absolutize_ref_uri(ref, parent_schema.uri)
       return true if self.class.schema_loaded?(schema_uri)
 
+      begin
+        validator = self.class.validator_for_uri(schema_uri)
+        schema_uri = JSON::Util::URI.file_uri(validator.metaschema)
+      rescue JSON::Schema::SchemaError
```

Comment:
Yes... That is exactly right. I tried to avoid refactoring the other method. Perhaps it should take an option, so that it can return nil if it finds nothing? (Or, even better, I could replace validator_for with two methods - validator_for and validator_for! - and have the first method return nil and the second throw an exception?)